pkg/slayers/path/scion/raw_spec.gobra

```diff
@@ -689,15 +689,9 @@ func (s *Raw) DecodingLemma(ubuf []byte, info path.InfoField, hop path.HopField)
 	hfIdxSeg := currHfIdx-prevSegLen
 	reveal s.CorrectlyDecodedInf(ubuf, info)
 	reveal s.CorrectlyDecodedHf(ubuf, hop)
-	pkt := reveal s.absPkt(ubuf)
-	currseg := reveal CurrSeg(ubuf, offset, currInfIdx, hfIdxSeg, segLen, MetaLen)
-	hopFields := hopFields(ubuf, offset, 0, segLen)
-	hopFieldsBytePositionsLemma(ubuf, offset, 0, segLen, R54)
-	reveal hopFieldsBytePositions(ubuf, offset, 0, segLen, hopFields)
-	assert currseg.Future[0] == hopFields[hfIdxSeg]
-	assert hopFields[hfIdxSeg] ==
-		path.BytesToIO_HF(ubuf, 0, offset + path.HopLen * hfIdxSeg, len(ubuf))
-	assert currseg.Future[0] == path.BytesToIO_HF(ubuf, 0, offset + path.HopLen * hfIdxSeg, len(ubuf))
+	reveal s.absPkt(ubuf)
+	reveal CurrSeg(ubuf, offset, currInfIdx, hfIdxSeg, segLen, MetaLen)
```

Comment:
I am increasingly concerned with these calls with `reveal`; in particular, I think they are hard to maintain because it is not clear which property we depend upon here. In the future, we should opt for explicit calls to lemmas that establish properties of the functions, rather than over-approximating it with a call to a revealed function